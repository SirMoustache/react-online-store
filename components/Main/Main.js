/**
 * Absolute imports
 */
import styled from 'styled-components';

const Main = styled.main`
  flex: 1 1 auto;
`;

export default Main;