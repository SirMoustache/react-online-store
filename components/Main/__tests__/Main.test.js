/**
 * Absolute imports
 */
import React from 'react';
import { render, cleanup } from '@testing-library/react';

/**
 * Local Components
 */
import Main from '..';

const child = <span className="unique">Test</span>;

const renderComponent = ({ ...rest } = {}) =>
  render(<Main {...rest}>{child}</Main>);

describe('<Main />', () => {
  afterEach(cleanup);

  it('should match snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('should render a <main> tag', () => {
    const { container } = renderComponent();
    expect(container.querySelector('main')).not.toBeNull();
  });

  it('should have a className attribute', () => {
    const className = 'test';
    const { container } = renderComponent({ className });
    expect(container.querySelector('main').classList).toContain(className);
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const { container } = renderComponent({ id });
    expect(container.querySelector('main').id).toContain(id);
  });

  it('should not adopt an invalid attribute', () => {
    const attribute = 'test';
    const { container } = renderComponent({ attribute });
    expect(
      container.querySelector('main').getAttribute('attribute'),
    ).toBeNull();
  });
});
