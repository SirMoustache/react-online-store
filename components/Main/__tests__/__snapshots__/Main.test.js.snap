// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should match snapshot 1`] = `
<div>
  <main
    class="Main-o02ej9-0 pINdQ"
  >
    <span
      class="unique"
    >
      Test
    </span>
  </main>
</div>
`;
